modules/xfeatures2d/src/freak.cpp

```diff
@@ -675,10 +669,10 @@ imgType FREAK_Impl::meanIntensity( InputArray _image, InputArray _integral,
     // expected case:
 
     // calculate borders
-    const int x_left = int(xf-radius+0.5);
-    const int y_top = int(yf-radius+0.5);
-    const int x_right = int(xf+radius+1.5);//integral image is 1px wider
-    const int y_bottom = int(yf+radius+1.5);//integral image is 1px higher
+    const int x_left = cvRound(xf-radius);
+    const int y_top = cvRound(yf-radius);
+    const int x_right = cvRound(xf+radius+1);//integral image is 1px wider
+    const int y_bottom = cvRound(yf+radius+1);//integral image is 1px higher
```

Comment:
Why not just cvFloor() / cvCeil()?

Reply:
Suppose when `xf-radius` is 2.9 and 2.1, it will be more sensible to assign 3 and 2 respectively in `x_left`.